sunbird_service_api_test/src/test/java/org/sunbird/integration/test/org/ReadOrganisationTest.java

```diff
@@ -62,4 +73,46 @@ public void testReadOrganisationFailure(String testName, HttpStatus httpStatusCo
         httpStatusCode,
         RESPONSE_JSON);
   }
+
+
+  @DataProvider(name = "readOrgSuccessDataProvider")
+  public Object[][] readOrgSuccessDataProvider() {
+
+    return new Object[][] {
+            new Object[] {TEST_NAME_READ_ORG_SUCCESS_WITH_VALID_ORG_ID, HttpStatus.OK},
+            new Object[] {TEST_NAME_READ_ORG_SUCCESS_WITH_VALID_PROVIDER_AND_EXTERNAL_ID, HttpStatus.OK}
+    };
+  }
+  @Test(dataProvider = "readOrgSuccessDataProvider")
+  @CitrusParameters({"testName", "httpStatusCode"})
+  @CitrusTest
+  public void testReadOrganisationSuccess(String testName, HttpStatus httpStatusCode) {
+    variable("rootExternalId", externalId);
+    beforeTest();
+    performPostTest(
+            this,
+            TEMPLATE_DIR,
+            testName,
+            getSearchOrgUrl(),
+            REQUEST_JSON,
+            MediaType.APPLICATION_JSON,
+            false,
+            httpStatusCode,
+            RESPONSE_JSON);
+  }
+
+  private void beforeTest() {
+    if(StringUtils.isBlank(orgId)){
+      getAuthToken(this, true);
+      OrgUtil.createOrg(
+              this,
+              testContext,
+              "templates/organisation/create",
+              "testCreateOrgSuccessWithProviderAndExternalId",
```

Comment:
testCreateRootOrgSuccessWithProviderAndExternalId

Check naming of other such tests.